x-pack/plugins/maps/public/layers/tiled_vector_layer.tsx

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { EuiIcon } from '@elastic/eui';
+import { IVectorStyle, VectorStyle } from './styles/vector/vector_style';
+import { SOURCE_DATA_ID_ORIGIN, LAYER_TYPE } from '../../common/constants';
+import { VectorLayer, VectorLayerArguments } from './vector_layer';
+import { canSkipSourceUpdate } from './util/can_skip_fetch';
+import { ITiledSingleLayerVectorSource, IVectorSource } from './sources/vector_source';
+import { SyncContext } from '../actions/map_actions';
+import { ISource } from './sources/source';
+import {
+  MapFilters,
+  VectorLayerDescriptor,
+  VectorSourceRequestMeta,
+} from '../../common/descriptor_types';
+
+export class TiledVectorLayer extends VectorLayer {
+  static type = LAYER_TYPE.TILED_VECTOR;
+
+  static createDescriptor(
+    descriptor: VectorLayerDescriptor,
+    mapColors: string[]
+  ): VectorLayerDescriptor {
+    const layerDescriptor = super.createDescriptor(descriptor, mapColors);
+    layerDescriptor.type = TiledVectorLayer.type;
+
+    if (!layerDescriptor.style) {
+      const styleProperties = VectorStyle.createDefaultStyleProperties(mapColors);
+      layerDescriptor.style = VectorStyle.createDescriptor(styleProperties);
+    }
+
+    return layerDescriptor;
+  }
+
+  readonly _source: ITiledSingleLayerVectorSource; // downcast to the more specific type
+
+  constructor({ layerDescriptor, source }: VectorLayerArguments) {
+    super({ layerDescriptor, source });
+    this._source = source as ITiledSingleLayerVectorSource;
+  }
+
+  getCustomIconAndTooltipContent() {
+    return {
+      icon: <EuiIcon size="m" type={this.getLayerTypeIconName()} />,
+    };
+  }
+
+  _getSearchFilters(
```

Comment:
Is this custom implementation even needed? Why not just use VectorLayer. _getSearchFilters?